libpromises/mod_custom.c

```diff
@@ -582,8 +582,36 @@ static void PromiseModule_Send(PromiseModule *module)
     DESTROY_AND_NULL(JsonDestroy, module->message);
 }
 
+static inline bool TryToGetContainerFromScalarRef(const EvalContext *ctx, const char *scalar, JsonElement **out)
+{
+    const size_t scalar_len = strlen(scalar);
+    if ((scalar[0] == '@') &&
+        (((scalar[1] == '(') && (scalar[scalar_len - 1] == ')')) ||
+         ((scalar[1] == '{') && (scalar[scalar_len - 1] == '}'))))
```

Comment:
what happens if `strlen` is 0 or 1 ?

Reply:
`scalar[0] == '\0'` in that case so the rest of the comparisons is skipped.

Reply:
Ah, you're right :+1: